ffs/types.go

```diff
@@ -342,6 +343,36 @@ func (cc ColdConfig) Validate() error {
 	return nil
 }
 
+type NotificationConfig struct {
+	Webhook       *Webhook
+	Configuration *WebhookConfiguration
+}
+
+type Webhook struct {
+	Endpoint       string
+	Authentication *WebhookAuthentication
+}
+
+type WebhookAuthentication struct {
+	Type string
+	Data *WebhookAuthData
+}
+
+type WebhookAuthData struct {
+	Username string
+	Password string
+}
+
+type WebhookConfiguration struct {
+	Events []string
+	Alerts []*WebhookAlert
+}
+
+type WebhookAlert struct {
+	Type      string
+	Threshold string
```

Comment:
Curious to see what you're thinking for `Threshold` use cases. Sounds interesting.